retrieval/db/ingest.py

```diff
@@ -481,40 +487,74 @@ def search_query_to_ingested_data(query: str, site_url: str, replace: bool = Fal
     return bool(scraped_data)
 
 
+def urls_to_ingested_data(
+    urls: List[str],
+    **kwargs,
+) -> None:
+    """Convert the webpages pointed to by urls into LangchainDocuments and ingest"""
+
+    docs = []
+    for url in urls:
+
+        logger.info(f"Scraping webpage {url}")
+
+        scraped_datum = scrape(url)
+
+        if scraped_datum and scraped_datum["text"]:
+            metadata = {k: v for k, v in scraped_datum.items() if k != "text"}
+            metadata["location"] = url
+            doc = LangchainDocument(page_content=scraped_datum["text"], metadata=metadata)
+            docs.append(doc)
+
+        else:
+            logger.info(f"Webpage {url} failed to scrape and/or did not seem to have any text")
+
+    if docs:
+        ingest(docs, **kwargs)
+
+    else:
+        logger.info("No docs to ingest!")
+
+
 if __name__ == "__main__":
 
     # SETTINGS
```

Comment:
Is the workflow at the moment, that I would need to change the hard-coded settings here and then run this script?

I would suggest adjusting this script to use command line variables - can use packages like argparse or typer for this purpose. This would make it easier to run the code in different modes without the need to hard code any settings (and instead the instructions can be given in the readme on what commands need to be ran).